solutions/gold/cses-1670.mdx

```diff
@@ -86,5 +96,99 @@ int main() {
 ```
 
 </CPPSection>
+</LanguageSection>
+</Spoiler>
+
+
+<Spoiler title="Solution 2">
+<Warning>
+While this solution is much easier to implement, it is extremely difficult to write an implementation that does not TLE in Java or Python. It may also even TLE in C++ if written inefficiently.
+</Warning>
+## Explanation
+Since there are only $9!$ states the grid can be in, we can use BFS to find the
+number of moves needed to reach the $\texttt{target}$ from the starting grid. We
+represent each grid as a string read left-to-right then top-to-bottom.
+
+For example, the grid in the testcase,
+```
+2 1 3
+7 5 9
+8 4 6
+```
+becomes "213759846".
 
+We then perform BFS normally but with a few notable changes. Since we cannot have
+our strings be the indices of an array, we store the "depth" of each string (number of moves required to reach a board configuration) inside our queue.
```

Comment:
```suggestion
We then perform BFS normally but with a few notable changes. First, since we can't have
strings be the indices of an array, we store the "depth" of each string (number of moves required to reach a board configuration) inside our queue.
```